lib/cache-activator.js

```diff
@@ -0,0 +1,53 @@
+const hasher = require('node-object-hash');
+const {startUpdater, cache} = require('./cache-manager');
+const CacheItem = require('./cache-item');
+const hashSha1 = hasher({alg: 'sha1'});
+
+let originalPromise = null;
+
+function augment(pp, fnName) {
+  const fetch = pp[fnName];
+  pp[fnName] = async function(...args) {
```

Comment:
Isn't this actually going to cache the results from more than just SELECT statements?  what about such things as
```
pp.execute(`DELETE table WHERE..`)
pp.execute(`UPDATE table ..`)
pp.execute(`INSERT into table..`)
```
Can you only apply this caching logic to statements that actually begin with "SELECT"?